aptos-move/move-examples/staking/sources/commission.move

```diff
@@ -0,0 +1,119 @@
+/// This contract is used to manage the commission rate for the node operator. There are two entities involved:
+/// 1. Manager: The account that can set the commission rate and change the operator account.
+/// 2. Operator: The account that receives the commission in dollars in exchange for running the node.
+///
+/// The commission rate is set in dollars and will be used to determine how much APT the operator receives.
+/// The commission is distributed to the operator and remaining amount to the manager. If there's not enough balance
+/// to pay the commission, either commission rate is set too high or APT price is low. In this case, the commission
+/// debt will be updated and the operator will receive the remaining balance in the next distribution.
+module staking::commission {
+    use aptos_framework::account::{Self, SignerCapability};
+    use aptos_framework::aptos_account;
+    use aptos_framework::aptos_coin::AptosCoin;
+    use aptos_framework::coin;
+    use aptos_framework::resource_account;
+    use aptos_framework::timestamp;
+    use aptos_std::math128;
+    use aptos_std::math64;
+    use staking::oracle;
+    use std::signer;
+
+    const INITIAL_COMMISSION_AMOUNT: u64 = 100000;
+    const ONE_YEAR_IN_SECONDS: u64 = 31536000;
+
+    /// Account is not authorized to call this function.
+    const EUNAUTHORIZED: u64 = 1;
+
+    struct CommissionConfig has key {
+        // The manager of the contract who can set the commission rate.
+        manager: address,
+        // The operator who receives the specified commission in dollars in exchange for running the node.
+        operator: address,
+        // The yearly commission rate in dollars. Will be used to determine how much APT the operator receives.
+        yearly_commission_amount: u64,
+        // Used to withdraw commission.
+        signer_cap: SignerCapability,
+        // Timestamp for tracking yearly commission.
+        last_update_secs: u64,
+        // Amount of debt in dollars owed to the operator due to insufficient amount received from node commission.
+        // This can happen if the commission rate is set too high or APT price is too low.
+        commission_debt: u64,
+    }
```

Comment:
I think doc comments end up being used in the doc generator.  Probably better to use that and have a good example

Reply:
Updated